monai/transforms/spatial/dictionary.py

```diff
@@ -517,7 +517,8 @@ def __call__(self, data: Mapping[Hashable, torch.Tensor], lazy: bool | None = No
                 output_spatial_shape=output_shape_k if should_match else None,
                 lazy=lazy_,
             )
-            output_shape_k = d[key].peek_pending_shape() if isinstance(d[key], MetaTensor) else d[key].shape[1:]
+            if output_shape_k is None:
```

Comment:
Hi @wyli, I just updated here to always refer to the output shape of the first key based on your comments [here](https://github.com/Project-MONAI/MONAI/issues/6911#issuecomment-1697843271).